File: ReadingList.Core/DataTransferObjects/BookDto.cs
namespace ReadingList.Core.DataTransferObjects;

public class BookDto
{
    public Guid Id { get; set; }
    public string Title { get; set; }

    public Guid AuthorId { get; set; }
    public AuthorDto Author { get; set; }

    public Guid CategoryId {get; set; }
    public CategoryDto Categories { get; set; }

    public BookNoteDto BookNote { get; set; }
}

File: ReactApp/reading-list-app/src/dto/book-dto.js
export default class BookDto {
  id = "";
  title = "";
  authorId = "";
  categoryId = "";

  constructor(id, title, authorId, categoryId) {
    this.id = id;
    this.title = title;
    this.authorId = authorId;
    this.categoryId = categoryId;
  }

  static fromResponse(response) {
    return new BookDto(
      response.id,
      response.title,
      response.authorId,
      response.categoryId
    );
  }
}
